File: Algorithms/Exponential Calculator/Python/ExponentialCalculator.py
def Factorial(n):
  r = 1;
  for i in range(n, 0, -1):
    r *= i;
  return r;

'''any number over 170 will return an error. Factorial are large numbers, 
floats can not handle it when it grows too much.
if you want try higher numbers please use decimal.Decimal() 
module (IT'S SLOWER THAN BASIC FLOAT!)
!170 already gives a very precise answer!'''
def Exponential(e):
  er = 1;
  for i in range(1, 170):
    er += ((e)**i)/Factorial(i);
  return er;

print(Exponential(1.5));

File: Algorithms/Exponential Calculator/ExponentialCalculator.cpp
#include <iostream>
#include <math.h>

double factorial(int n)
{
	long double r = 1;
	for(int i = n; i > 0; i--)
	{
		r *= i;
	}
	return r;
}

double exponential(double e)
{
  double er = 1.0;
  for(int i = 1; i < 170; i++)
  {
    er += pow(e, i)/factorial(i);
  }
  return er;
}

int main()
{
	std::cout << exponential(1.5) << std::endl;
	return 0;
}